aiida/orm/implementation/general/calculation/job/__init__.py

```diff
@@ -37,6 +36,29 @@ class AbstractJobCalculation(object):
     remotely on a job scheduler.
     """
 
+    _cacheable = True
+
+    @classproperty
+    def _hash_ignored_attributes(cls):
+        # _updatable_attributes are ignored automatically.
+        return super(AbstractJobCalculation, cls)._hash_ignored_attributes + [
+            'queue_name',
+            'max_wallclock_seconds',
```

Comment:
For `queue_name` and `max_wallclock_seconds`, while I see the reason for not caching, is this really something we want to do? If yes, then for consistency we should also ignore the other things that are setting properties on the scheduler? 
```
        custom_sched_commands = self.get_custom_scheduler_commands()
        if custom_sched_commands:
            job_tmpl.custom_scheduler_commands = custom_sched_commands

        job_tmpl.import_sys_environment = self.get_import_sys_environment()

        job_tmpl.job_environment = self.get_environment_variables()

        queue_name = self.get_queue_name()
        if queue_name is not None:
            job_tmpl.queue_name = queue_name
        priority = self.get_priority()
        if priority is not None:
            job_tmpl.priority = priority
        max_memory_kb = self.get_max_memory_kb()
        if max_memory_kb is not None:
            job_tmpl.max_memory_kb = max_memory_kb
        max_wallclock_seconds = self.get_max_wallclock_seconds()
        if max_wallclock_seconds is not None:
            job_tmpl.max_wallclock_seconds = max_wallclock_seconds
        max_memory_kb = self.get_max_memory_kb()
        if max_memory_kb is not None:
            job_tmpl.max_memory_kb = max_memory_kb
```
(this comes from the ```JobCalculation._presubmit()``` method).
On the other hand, some of these flags might change the actual behaviour of the code (crash if not enough memory, run with different options if environment variables are different, ...). What is a good compromise? 

In any case, in the docs this choice is not documented, so maybe the info on what is used for the various classes (like: data -> everything; JobCalcs: this above; InlineCalcs: inputs; other Calcs: not cached; ...) should be explained to the user. (and maybe for a subclass how to change this - maybe for a plugin you might want to add more?)

Reply:
As long as the only possible change is that a job can fail if the setting is wrong (e.g. time limit too low) there shouldn't be a problem, because failed calculations are anyway not taken into account when using the cache. Environment variables are a different story because they might change the result, but there's also a grey area, where a calculation _could_ behave differently if e.g. the memory limit is different, but that doesn't strike me as very realistic.

So maybe it's ok to add ``'max_memory_kb'`` and ``'priority'`` to the ignore list? Or would you rather go the conservative way of not ignoring any of them?

Obviously this should be noted in the documentation.

Reply:
If we keep the other things like the queue, then ok to add also max_memory_kb and priority I think.

Reply:
Ok, I added them now. However, I'm now having doubts whether it's a good idea to put this in the documentation. My concern is that when this is inevitably changed in the future, the documentation will most probably not reflect that change. I don't think there's a simple way to keep it up-to-date.

What do you think? Should I put it in anyway?